pkg/cli/kubernetes/portforward/application_watcher.go

```diff
@@ -60,16 +58,21 @@ func NewApplicationWatcher(options Options) *applicationWatcher {
 func (aw *applicationWatcher) Run(ctx context.Context) error {
 	defer close(aw.done)
 
-	// We use the `radapp.io/application` label to include pods that are part of an application.
-	// This can include the user's Radius containers as well as any Kubernetes resources that are labeled
-	// as part of the application (eg: something created with a recipe).
-	req, err := labels.NewRequirement(kubernetes.LabelRadiusApplication, selection.Equals, []string{aw.Options.ApplicationName})
+	deployments := aw.Options.Client.AppsV1().Deployments(aw.Options.Namespace)
+	listOptions := metav1.ListOptions{LabelSelector: aw.Options.LabelSelector.String()}
+
+	// List all deployments that match the label selector
+	labelledDeployments, err := deployments.List(ctx, listOptions)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to list deployments: %w", err)
 	}
 
-	deployments := aw.Options.Client.AppsV1().Deployments(aw.Options.Namespace)
-	listOptions := metav1.ListOptions{LabelSelector: labels.NewSelector().Add(*req).String()}
+	if len(labelledDeployments.Items) == 0 {
+		dashboardLabels := CreateLabelsForDashboard()
+		if aw.Options.LabelSelector.Matches(dashboardLabels) {
+			fmt.Println("Radius Dashboard not found, please see https://docs.radapp.io/guides/tooling/dashboard for more information")
```

Comment:
Is there a better option here than using `fmt.Println`? I try to avoid using `fmt.Println` anywhere in CLI code because it's inflexible. 

I think it would also be better to augment the `Options` to have a "warn if no deployments found" feature. This hardcodes the dashboard functionality in the middle of the code.

Reply:
yeah I don't really like putting this here either - maybe we could move the dashboard check up a bit and remove this custom functionality from application_watcher. I will try it

Reply:
Newest commit moves the logic to run instead of applicationWatcher.